tests/testthat/test-validate_submission.R

```diff
@@ -319,3 +319,31 @@ test_that("validate_submission handles overriding output type id data type corre
     )[["col_types"]]
   )
 })
+
+test_that("Ignoring derived_task_ids in validate_submission works", {
```

Comment:
similar question to in another test -- to test whether `derived_task_ids` were actually ignored, it seems like a test would involve setting a derived task id to something invalid and then seeing that an error was not raised?

Reply:
I actually think maybe the question is not relevant where I asked it first, but may be relevant here.

Reply:
Great points about the tests! I sort of knew from the benchmarks that it was working but admittedly, this wasn't being that explicitly tested throughout (mainly just in the `expand_model_out_grid()` tests.

So added a bunch of tests throughout following your suggestion of introducing a deliberate error. 23b4655107f89300b8ca23350d516bb96c4b5917

Reply:
23b4655107f89300b8ca23350d516bb96c4b5917